src/utils/gameUtils.js

```diff
@@ -0,0 +1,97 @@
+// if we need a list of keys
+// const alpha = Array.from(Array(lenAlpha)).map((e, i) => i + offsetAlpha);
+// const alphabet = alpha.map((x) => String.fromCharCode(x));
+const letterDistribution = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1
+};
+export const createInitialLetterBag = () => {
+  // let initialLetterBag = [];
+  // for (const [letter, count] of Object.entries(letterDistribution)) {
+  // 	initialLetterBag.push([letter, count]);
+  // }
+  return letterDistribution;
+};
+
+// Function to draw a random letter from the letter bag into the hand
+export const drawRandomLetter = (letterBag) => {
+  // const letterCounts = Object.values(letterBag);
+  // letterCounts.sort((a, b) => {
+  //   return b - a;
+  // }); // sort in descending order
+  // const highestCount = letterCounts[0]; // grab first item (highest count)
+  // if (highestCount === 0) {
+  //   return null; // all letters spent and bag is empty (change the draw message?)
+  // }
+  // const randomChar = String.fromCharCode(Math.floor(Math.random() * lenAlpha) + offsetAlpha);
+  // if (letterBag[randomChar] === 0) {
+  //   return drawRandomLetter(letterBag); // try again (end case handled above when all letter counts are zero)
+  // } else {
+  //   return randomChar;
+  // }
+  const availableLetters = Object.entries(letterBag).reduce(
+    (l, [key, value]) => (value > 0 ? { ...l, [key]: value } : l),
+    {}
+  );
+  const letters = Object.keys(availableLetters);
+  const count = letters.length;
+  const idx = Math.floor(Math.random() * count);
+  const letter = letters[idx];
+  return letter;
```

Comment:
- only consider available letters - avoid having to call multiple times (line 52)
- line 47 is taken care of inside reducer. once all letters have been drawn, `canDraw` gets set to false and button gets disabled